crates/bitcoin/src/types.rs

```diff
@@ -319,6 +321,12 @@ impl Transaction {
         sha256d_le(&self.format_with(false))
     }
 
+    pub fn tx_id_bounded(&self, length_bound: u32) -> Result<H256Le, Error> {
+        let bytes = self.format_with(false);
```

Comment:
You can set `length_bound` and still execute all of `format_with` meaning the weight could get underestimated